source/uri-options/tests/compression-options.yml

```diff
@@ -0,0 +1,44 @@
+tests:
+    -
+        description: "Valid compression options are parsed correctly"
+        uri: "mongodb://example.com/?compressors=zlib&zlibCompressionLevel=9"
+        valid: true
+        warning: false
+        hosts: ~
+        auth: ~
+        options:
+            compressors: "zlib"
+            zlibCompressionLevel: 9
+    -
+        description: "Invalid compressors cause a warning"
```

Comment:
The Connection String spec specifically mandates that drivers should warn rather than error on invalid options (see the last part of [this section](https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst#values) for the part about values and the third question [here](https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst#q-a) for the rationale. This spec is specifically designed to be compatible with the Connection String spec, so we followed those guidelines.

Reply:
Thanks.  It looks like the Java driver is inconsistent about this.  It warns for certain things like booleans with odd values, but errors for others, like integers.  We'll fix it up.